File: app/src/main/java/com/example/jackknife/bean/Menu.java
package com.example.jackknife.bean;

import androidx.databinding.BaseObservable;

public class Menu extends BaseObservable {

    int id;
    String title;
    String path;

    public Menu(int id, String title, String path) {
        this.id = id;
        this.title = title;
        this.path = path;
    }

    /**
     * {@link android.widget.TextView#setText(int)} )}时，如果给这个id，会招不到值，报错，因为这个id不是资
     * 源id。
     *
     * @return
     */
    public String getId() {
        return String.valueOf(id);
    }

    public String getTitle() {
        return title;
    }

    public String getPath() {
        return path;
    }
}

File: app/src/main/java/com/example/jackknife/bean/Menu.kt
package com.example.jackknife.bean

import androidx.databinding.BaseObservable
import androidx.databinding.Bindable

data class Menu(var id: Int, var title: String, var path: String) : BaseObservable() {

//    /**
//     * [android.widget.TextView.setText] )}时，如果给这个id，会找不到值，报错，因为这个id不是资
//     * 源id。
//     *
//     * @return
//     */
//    @Bindable
//    fun getId(): String {
//        return id.toString()
//    }

    /**
     * 兼容CircleCI的docker镜像的编译。
     *
     * @return
     */
    @Bindable
    fun getIdToString(): String {
        return id.toString()
    }
}